src/NHibernate/Dialect/Function/SQLFunctionRegistry.cs

```diff
@@ -20,8 +22,11 @@ public SQLFunctionRegistry(Dialect dialect, IDictionary<string, ISQLFunction> us
 		/// </summary>
 		public ISQLFunction FindSQLFunction(string functionName)
 		{
-			ISQLFunction result;
-			if (!userFunctions.TryGetValue(functionName, out result))
+			if (!userFunctions.ContainsKey(functionName) && !dialect.Functions.ContainsKey(functionName) && _functionAliases.TryGetValue(functionName, out var sqlFunction))
+			{
+				functionName = sqlFunction;
+			}
+			if (!userFunctions.TryGetValue(functionName, out ISQLFunction result))
 			{
 				dialect.Functions.TryGetValue(functionName, out result);
 			}
```

Comment:
This incurs double look-ups on the nominal case. I would rather avoid that.

```suggestion
			if (!userFunctions.TryGetValue(functionName, out ISQLFunction result) && !dialect.Functions.TryGetValue(functionName, out result))
			{
				if (_functionAliases.TryGetValue(functionName, out var sqlFunction) && !_functionAliases.ContainsKey(sqlFunction))
					return FindSQLFunction(sqlFunction);
			}
```

The ContainsKey is there to prevent an endless recursion in case of a badly setup `_functionAliases`.